Sources/Apodini/Modifier/DelegationModifier.swift

```diff
@@ -0,0 +1,146 @@
+//
+//  DelegatingHandlerInitializer.swift
+//  
+//
+//  Created by Max Obermeier on 06.06.21.
+//
+
+import Foundation
+import ApodiniUtils
+
+// MARK: Component+DelegationModifier
+
+extension Component {
+    /// Use a `DelegatingHandlerInitializer` to create a fitting delegating `Handler` for each of the `Component`'s endpoints.
+    /// All instances created by the `initializer` can delegate evaluations to their respective child-`Handler` using `Delegate`.
+    /// - Parameter prepend: If set to `true`, the modifier is prepended to all other calls to `delegated` instead of being appended as usual.
+    /// - Note: `prepend` should only be used if `I.Response` is `Self.Response` or `Self` is no `Handler`.
+    public func delegated<I: DelegatingHandlerInitializer>(by initializer: I, prepend: Bool = false) -> DelegationModifier<Self, I> {
+        DelegationModifier(self, initializer: initializer, prepend: prepend)
+    }
+}
+
+// MARK: DelegationModifier
+
+public struct DelegationModifier<C: Component, I: DelegatingHandlerInitializer>: Modifier {
+    public typealias ModifiedComponent = C
+    
+    public let component: C
+    private let initializer: I
+    private let prepend: Bool
+    
+    public var content: some Component { EmptyComponent() }
```

Comment:
Is there a reason why this Modifier adds the content property?
Modifier sets the `Content` type alias to Never if I'm not wrong? 🤔

Reply:
You're right, that doesn't make sense. It doesn't really have an effect either, though. I guess I just copied that part form `ResponseTransformer` or `Guard` stuff...